debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSignalBasedIncrementalSnapshotChangeEventSource.java

```diff
@@ -57,4 +68,155 @@ protected Table refreshTableSchema(Table table) throws SQLException {
         schema.refresh(jdbcConnection, table.id(), true);
         return schema.tableFor(table.id());
     }
+
+    @Override
+    protected String buildChunkQuery(Table table, int limit, Optional<String> additionalCondition) {
+        // Select query builder based on the configuration only on PostgreSQL Connector (incremental.snapshot.query.builder.type)
+        if (postgresConnectorConfig.getIncrementalSnapshotQueryBuilderType() == PostgresConnectorConfig.IncrementalSnapshotChunkQueryBuilderType.UNION_BASED) {
+            return buildUnionBasedChunkQuery(table, limit, additionalCondition);
+        }
+
+        return super.buildChunkQuery(table, limit, additionalCondition);
+    }
+
+    @Override
+    protected PreparedStatement readTableChunkStatement(String sql) throws SQLException {
+        final PreparedStatement statement = jdbcConnection.readTablePreparedStatement(connectorConfig, sql,
+                OptionalLong.empty());
+        if (context.isNonInitialChunk()) {
+            final Object[] maximumKey = context.maximumKey().get();
+            final Object[] chunkEndPosition = context.chunkEndPosititon();
+            // Fill boundaries placeholders
+            int pos = 0;
+            final List<Column> queryColumns = getQueryColumns(getCurrentTable());
+
+            if (postgresConnectorConfig.getIncrementalSnapshotQueryBuilderType().equals(PostgresConnectorConfig.IncrementalSnapshotChunkQueryBuilderType.SIMPLE)) {
+                for (int i = 0; i < chunkEndPosition.length; i++) {
+                    for (int j = 0; j < i + 1; j++) {
+                        jdbcConnection.setQueryColumnValue(statement, queryColumns.get(j), ++pos, chunkEndPosition[j]);
+                    }
+                }
+                // Fill maximum key placeholders
+                for (int i = 0; i < chunkEndPosition.length; i++) {
+                    for (int j = 0; j < i + 1; j++) {
+                        jdbcConnection.setQueryColumnValue(statement, queryColumns.get(j), ++pos, maximumKey[j]);
+                    }
+                }
+            }
+            else if (postgresConnectorConfig.getIncrementalSnapshotQueryBuilderType()
```

Comment:
Do we have a unit test for this part?

Reply:
IIRC, the method requires a DB connection; I haven't added a unit test for it.